### PR TITLE
Modernize the plugin

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Modernize the plugin

Incomplete set of changes that may help a future maintainer of this plugin as they modernize it and test it interactively for eventual release.

Require Jenkins 2.414.3 or newer.

Use most recent parent pom.

Compiles and passes tests with Java 21 on Linux.  Compiles and passes tests with Java 17 on Windows.

Powermock tests have been removed.  Powermock is dead and would need to be replaced with Mockito.

Mockito tests have been removed.  Mockito references were outdated.  I'm not passionate enough about Mockito based tests to adapt them.  If a new maintainer wants to bring back those tests, they are in the source history.  

Selenium tests have been removed.  Jenkins user interface has changed significantly enough since this plugin was last maintained that I expect all the Selenium tests will fail for reasons that are unrelated to plugin functionality.  I'm not passionate enough for Selenium testing to do the necessary work to adapt the tests to the current Jenkins behavior.  If a new maintainer wants to bring back those tests, they are in the source history.

Supersedes existing pull requests:

* #6 
* #5 
* #4 
* #3 
* #2 
* #1 

### Testing done

Ran the plugin automated tests and confirmed they pass.  Suppressed the spotbugs warnings so that they pass.  Ran the plugin interactively with Jenkins 2.440.1 and confirmed that it started as expected.

I'm not a user of the plugin, so I didn't perform any interactive verification of functionality.

The dependency on the deprecated analysis plugin has not been removed.  That's left to someone else to decide how to handle that.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MarkEWaite** [comment]
@NotMyFault if you're feeling bold, this can be merged to the master branch and all the open dependabot pull requests can be closed.  I've not tested it interactively, so I can't be confident that it is functional, but I expect it to be functional

**Comment by NotMyFault** [comment]
> @NotMyFault if you're feeling bold, this can be merged to the master branch and all the open dependabot pull requests can be closed. I've not tested it interactively, so I can't be confident that it is functional, but I expect it to be functional

@rzanner adopted the plugin recently, I think he can take care of this :)

**Review by rzanner — APPROVED** [review]
Thanks for the effort, @MarkEWaite! I guess we can merge. :)

**Comment by bignay2000** [comment]
@rzanner @MarkEWaite  - I am a power user of this plugin and I have complex pipelines.  So given the hpi file I can load that into my dev Jenkins and give some feedback.  How do we go about a build process to get the HPI official built from Master branch?

**Comment by bignay2000** [comment]
1. Successfully ran tests and built Master branch via openjdk 17.  
2. Successfully created the hpi - delivery-pipeline-plugin-1.4.3-SNAPSHOT
3. Successfully loaded hpi file and restarted Jenkins
4. Successfully ran a complex pipeline! :
a. Job Name correctly displayed
b. Build Number correctly displayed
c. Arrows are correctly displayed
d. Red color correctly displays on failure
e. Blue color correctly displays
f. Green color correctly displays on success
g. Progress bar correctly displays
h. Hyperlinks work
I. Job History works
J. Build button icon does not load (this is also broken in the last 1.42 release)


**Build Environment**
Macbook Pro M1 Pro
MacOS 14.6.1

Homebrew (https://brew.sh)
`
brew install openjdk@17
export JAVA_HOME="/Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home"
cd ../git/delivery-pipeline-plugin

mvn clean install -DskipTests  

mvn clean install

mvn hpi:hpi`

../git/delivery-pipeline-plugin/target/delivery-pipeline-plugin.hpi

**Environment**
Hypervisor: Proxmox
VM: Alpine Linux
Docker: jenkins/jenkins:2.462.1-alpine-jdk21

**Comment by bignay2000** [comment]
Build Now Button is still missing, which is broken in the current 1.42 release...   But otherwise Master branch looks to be fully working.

Screenshot 2024-09-03 at 6 00 47 PM

**Comment by bignay2000** [comment]
Going to do more testing tomorrow

1. Test a pipeline with a manual step
2. Confirm retry works with the retry icon

**Comment by basil** [comment]
> I'm not passionate enough for Selenium testing to do the necessary work to adapt the tests to the current Jenkins behavior.

But those tests are important to verify that future changes do not regress the UI for this JavaScript-heavy plugin.

File: pom.xml [hunk 1]
```diff
@@ -37,448 +37,45 @@
...
-                            <goal>check</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-checkstyle-plugin</artifactId>
-                <configuration>
-                    <configLocation>checkstyle-rules.xml</configLocation>
```
Comment (basil) [inline pom.xml:255]: @MarkEWaite Why was this Checkstyle configuration deleted from `pom.xml` without deleting the corresponding `checkstyle-rules.xml` file from the repository?

Comment (MarkEWaite) [reply pom.xml:255]: A mistake on my part.  I can submit a pull request to remove the checkstyle rules file or can submit a pull request to restore the checkstyle configuration.  Which do you prefer?

Comment (basil) [reply pom.xml:255]: No strong preference, as long as the state of the repository is consistent at any given point in time.

**Pull request merged**